modules/account-lib/src/coin/hbar/transactionBuilderFactory.ts

```diff
@@ -1,20 +1,81 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
+import { InvalidTransactionError, ParseTransactionError } from '../baseCoin/errors';
 import { BaseTransactionBuilderFactory } from '../baseCoin';
 import { WalletInitializationBuilder } from './walletInitializationBuilder';
 import { TransferBuilder } from './transferBuilder';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+import { isValidRawTransactionFormat, toUint8Array } from './utils';
 
 export class TransactionBuilderFactory extends BaseTransactionBuilderFactory {
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
 
   /** @inheritdoc */
-  getWalletInitializationBuilder(): WalletInitializationBuilder {
-    return new WalletInitializationBuilder(this._coinConfig);
+  getWalletInitializationBuilder(tx?: Transaction): WalletInitializationBuilder {
+    return this.initializeBuilder(tx, new WalletInitializationBuilder(this._coinConfig));
   }
 
   /** @inheritDoc */
-  getTransferBuilder(): TransferBuilder {
-    return new TransferBuilder(this._coinConfig);
+  getTransferBuilder(tx?: Transaction): TransferBuilder {
+    return this.initializeBuilder(tx, new TransferBuilder(this._coinConfig));
+  }
+
+  /** @inheritDoc */
+  from(raw: Uint8Array | string): TransactionBuilder {
+    this.validateRawTransaction(raw);
+    const tx = this.parseTransaction(raw);
+    switch (tx.txBody.data) {
+      case 'cryptoTransfer':
+        return this.getTransferBuilder(tx);
+      case 'cryptoCreateAccount':
+        return this.getWalletInitializationBuilder(tx);
+      default:
+        throw new InvalidTransactionError('Invalid transaction');
```

Comment:
it's possible we feed this entirely legit transactions and would like to know what type I'd think. maybe let's try to give it the `txBody.data` in the throw?

Reply:
Actually it's not posible. `txBody.data` will always be one of Cryptotranfer or CryptoCreate. We need to include this default because protobuff adds an undefined return value to all class variables